From: wuyifeng <wuyifeng@nd.com.cn>
Date: Thu, 15 Dec 2022 14:07:07 +0800
Subject: Hide ntp bookmark button when not login

Fix version: 1.2.1
Hide the new tab page's 'Bookmarks' when the user doesn't sign in to Promethean.
Relate to AP9-11067

---
 .../chromium/chrome/browser/ntp/NewTabPageLayout.java  | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/chrome/android/java/src/org/chromium/chrome/browser/ntp/NewTabPageLayout.java b/chrome/android/java/src/org/chromium/chrome/browser/ntp/NewTabPageLayout.java
--- a/chrome/android/java/src/org/chromium/chrome/browser/ntp/NewTabPageLayout.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/ntp/NewTabPageLayout.java
@@ -28,6 +28,7 @@ import androidx.annotation.VisibleForTesting;
 import org.chromium.base.Callback;
 import org.chromium.base.CallbackController;
 import org.chromium.base.MathUtils;
+import org.chromium.base.PrometheanManager;
 import org.chromium.base.TraceEvent;
 import org.chromium.chrome.R;
 import org.chromium.chrome.browser.app.video_tutorials.NewTabPageVideoIPHManager;
@@ -865,8 +866,13 @@ public class NewTabPageLayout extends LinearLayout implements TileGroup.Observer
         ViewStub shortcutsStub = findViewById(R.id.shortcuts_stub);
         mShortcutsView = (ViewGroup) shortcutsStub.inflate();

-        mShortcutsView.findViewById(R.id.bookmarks_button)
-                .setOnClickListener(view -> mManager.getNavigationDelegate().navigateToBookmarks());
+        View bookmarksButton = mShortcutsView.findViewById(R.id.bookmarks_button);
+        final boolean isLogin = PrometheanManager.INSTANCE.isLogin();
+        if (!isLogin) {
+            bookmarksButton.setVisibility(View.GONE);
+        } else {
+            bookmarksButton.setOnClickListener(view -> mManager.getNavigationDelegate().navigateToBookmarks());
+        }

         mShortcutsView.findViewById(R.id.downloads_button)
                 .setOnClickListener(
--
2.34.1

